esp-storage/src/buffer.rs

```diff
@@ -0,0 +1,61 @@
+use core::{mem::MaybeUninit, slice};
+
+#[cfg(feature = "nor-flash")]
+pub fn uninit_slice(bytes: &[u8]) -> &[MaybeUninit<u8>] {
+    unsafe { core::mem::transmute(bytes) }
+}
+
+#[cfg(feature = "nor-flash")]
+pub fn uninit_slice_mut(bytes: &mut [u8]) -> &mut [MaybeUninit<u8>] {
+    unsafe { core::mem::transmute(bytes) }
+}
+
+pub type FlashWordBuffer = FlashBuffer<4, 1>;
+pub type FlashSectorBuffer = FlashBuffer<4096, 1024>;
+
+#[repr(C)]
+pub union FlashBuffer<const N: usize, const M: usize> {
+    bytes: [MaybeUninit<u8>; M],
+    words: [MaybeUninit<u32>; N],
```

Comment:
Maybe I'm missing something but is the `words` field ever really used? (Yes, by the accessors but those seem to be unused?)

Reply:
I don't think specifying the alignment is needed according to this: https://doc.rust-lang.org/reference/type-layout.html#reprc-unions

Reply:
> Maybe I'm missing something but is the `words` field ever really used? (Yes, by the accessors but those seem to be unused?)

I have a following PR coming that uses those. I split in two to make more focused PR.